src/CakeSlayer/Measure.hs

```diff
@@ -43,48 +44,36 @@ type PrometheusRegistry = IORef (HashMap Text Histogram)
 -- | Constraint for the monadic actions that have access to measurements.
 type MonadMeasure m = (HasCallStack, MonadTimed m)
 
-{- | Helper function to be used in instance implementations.
+{- | Basic implementation of 'timeAction' function for 'MonadTimed' instances.
 
 It measures the time taken to perform the given action and store it
 in the @timings@ distribution with the label of the action.
 
-This is a recommended implementation which has the integration with @prometheus@.
+It also receives any function that work with metrics. This function can use
+label and time data inside.
 -}
-timedActionImpl
+timedActionSimpleImpl
     :: forall r m a .
        ( MonadReader r m
        , Has Timings r
        , Has Metrics.Store r
-       , Has PrometheusRegistry r
        , MonadIO m
        , HasCallStack
-       , MonadMonitor m
        )
-    => Text  -- ^ Global name
+    => (Text -> Double -> m ())  -- ^ any given metrics collection actions.
```

Comment:
Nice solution! I've been thinking about the solution as well. What do you think if this argument will become a list of functions so users can arbitrary and remove any functions? Something like this:

```haskell
timedActionWith [ekgTimings]
timedActionWith [prometheusTimings]
timedActionWith [ekgTimings, prometheusTimings]
```

Does the interface look reasonable?

Reply:
I'm not sure about the list, this actions are monadic so can be binded together...
I renamed and improved the docs a bit though.